src/core/feed-core.js

```diff
@@ -227,43 +237,49 @@ function _actionToFeedObject(row, client) {
     throw new Error('Client information not passed as a parameter');
   }
 
-  var feedObj = {
-    id: row['id'],
-    type: row['action_type_code'],
-    votes: row['votes'],
-    userVote: row['user_vote'],
-    hotScore: row['hot_score'],
-    author: {
-      id: row['user_id'],
-      name: row['user_name'],
-      team: row['team_name'],
-      type: _resolveAuthorType(row, client)
-    },
-    createdAt: row['created_at']
-  };
-
-  if (row.location) {
-    feedObj.location = {
-      latitude: row.location.y,
-      longitude: row.location.x
+// _getNumberOfComments calculates the number of comments for a feedItem
+  return _getNumberOfComments(row['id'], row['parent_id'], client)
+  .then(numberOfComments => {
+    var feedObj = {
+      id: row['id'],
+      type: row['action_type_code'],
+      votes: row['votes'],
+      userVote: row['user_vote'],
+      hotScore: row['hot_score'],
+      author: {
+        id: row['user_id'],
+        name: row['user_name'],
+        team: row['team_name'],
+        type: _resolveAuthorType(row, client)
+      },
+      createdAt: row['created_at'],
+      parent_id: row['parent_id'],
+      numberOfComments: numberOfComments,
     };
-  }
 
-  if (feedObj.type === 'IMAGE') {
-    const imagePath = row['image_path'];
+    if (row.location) {
+      feedObj.location = {
+        latitude: row.location.y,
+        longitude: row.location.x
+      };
+    }
+
+    if (feedObj.type === 'IMAGE') {
```

Comment:
Split this into a generic function and use it, since this isn't the only place where this logic exists.